src/MOCKernel.cpp

```diff
@@ -0,0 +1,165 @@
+#include "MOCKernel.h"
+#include "TrackGenerator.h"
+
+/**
+ * @brief Constructor for the MOCKernel assigns default values
+ * @param track_generator the TrackGenerator used to pull relevant tracking
+ *        data from
+ * @param row_num the row index into the temporary segments matrix
+ */
+MOCKernel::MOCKernel(TrackGenerator* track_generator, int row_num) {
+  _count = 0;
```

Comment:
You might even consider renaming `_count` since it isn't obvious what it represents